crates/wasmtime/src/runtime/values.rs

```diff
@@ -689,13 +702,25 @@ impl Ref {
                 );
                 Ok(TableElement::FuncRef(f.vm_func_ref(store).as_ptr()))
             }
+
+            #[cfg(feature = "gc")]
             (Ref::Extern(e), HeapType::Extern) => match e {
                 None => {
                     assert!(ty.is_nullable());
                     Ok(TableElement::ExternRef(None))
                 }
                 Some(e) => Ok(TableElement::ExternRef(Some(e.inner))),
             },
+
+            #[cfg(not(feature = "gc"))]
+            (Ref::Extern(e), HeapType::Extern) => match e {
+                None => bail!(
+                    "support for externref tables disabled at compile time \
+                     because the `gc` cargo feature was not enabled"
+                ),
+                Some(e) => match e._inner {},
+            },
```

Comment:
I realize that this is what I recommended, having an uninhabited public type, but I'm second guessing myself now. This feels like it's pushing a lot of `#[cfg]` noise into the consumers of externref.

Could the methods on `ExternRef`, at least used by Wasmtime itself, be preserved? That way only `ExternRef` itself needs `#[cfg]` and consumers don't have to mind whether it's defined or not.